Integration with snowfork bridge

yrong opened the pull request.

herryho commented on node/primitives/src/currency.rs:
```diff
@@ -187,7 +190,8 @@ macro_rules! create_currency_id {
 						(((*token_symbol_2 as u64) << 32) & 0x0000_00ff_0000_0000) + (((*token_type_2 as u64) << 40) & 0x0000_ff00_0000_0000) + discr
-					}
+					},
+					_ => 0 as u64,
```
0 represents CurrencyId::Native(TokenSymbol::ASG). We should not provide a wildcard here as each one of bifrost currencies needs to have a u64 presentation.

Since the discriminant for Erc20 is 7, we can compose an Erc20 currencyId like (7 <<8 + x), while x means tokensymbol related to the Erc20 token.

herryho replied:
x needs not to be the lowest 8 bits. Instead, it can make use of the upper 48 bits, as long as it keeps  the 7 discriminant in the right place.

herryho commented on node/primitives/src/currency.rs:
```diff
@@ -221,6 +226,7 @@ macro_rules! create_currency_id {
 					CurrencyId::LPToken(_ts1, _, _ts2, _) => {
 						stringify!(_ts1, ",", _ts2)
 					}
+					_ => "Unknown",
```
Is it possible that we maintain a map between Erc20 token list and bifrost TokenSymbol list? Might it be easier to see and convert?

yrong replied:
@herryho  Refact to use TokenSymbol mapping to Erc20 directly, as @lurpis mentioned the only Erc20 token we published in Ethereum Mainnet is 
https://etherscan.io/token/0xc3d088842dcf02c13699f936bb83dfbbc6f721ab, we just map it to VETH

yrong committed:
Add snowbridge into asgard

> just clone and link snowbridge into our repo
> add bridge related pallets into asgard

yrong committed:
ignore fmt

yrong committed:
Snowbridge integration-(to be continued)

yrong committed:
Revamp asgard spec for local test with geth
